### DISCUSSION TITLE
Status of HTML tables as input format

### DISCUSSION THREAD

**Discussion by baptiste** [quarto-dev/quarto-cli#5574] [Q&A] [answered]
I remember seeing [this 1.3 pre-release announcement](https://quarto.org/docs/prerelease/1.3/tables.html) that Quarto would enable HTML tables as input format, affording more flexibility compared to the various markdown idioms.

I wanted to test this today with this rather unwieldy table that I find really hard to edit, let alone format, in markdown:

![Screenshot 2023-05-17 at 07 14 43](https://github.com/quarto-dev/quarto-cli/assets/18970/64538843-bdbd-4be3-870e-cadd4c14ee1d)

I tried the suggested HTML format, but that doesn't seem to be parsed in Quarto v1.3.353, and indeed I cannot find any mention of it in the current documentation. Was this feature removed for now? Any chance it might come back?

**Comment by mcanouil** [comment]
Could you share a small "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by cscheid** [comment]
> I tried the suggested HTML format, but that doesn't seem to be parsed in Quarto v1.3.353, and indeed I cannot find any mention of it in the current documentation.

We haven't done a great job moving the documentation from the pre-rerelease descriptions to the webpages themselves, and we'll be fixing that shortly.


> Was this feature removed for now?

It hasn't been removed - can you provide an example where it doesn't work for you? It's possible that the table you're trying to write is not valid HTML. If it validates correctly here: https://validator.w3.org/ and it doesn't show up how you expect it to, then we would consider that a bug.

**Comment by baptiste** [comment]
I think I'm just not understanding the syntax – even the example from the pre-release page doesn't work for me, so I'm guessing it needs to be wrapped in a container of some sort?

````
---
title: "testing table"
format: 
   html:
     keep-md: true
   pdf:
     keep-tex: true
---


<table>
<caption><span data-qmd="As described in @Lovelace1864, computers are great."></span></caption>
<thead>
<tr>
<th><span data-qmd="_Header 1_"></span></th>
<th><span data-qmd="_Header 2_"></span></th>
</tr>
</thead>
<tbody>
<tr>
<td><span data-qmd=""></span></td>
<td>Regular output</td>
</tr>
</tbody>
</table>

````

**Reply by cscheid** [reply]
You need an explicit raw html block, or Pandoc attempts to parse this differently:

````
```{=html}
<table>
<caption><span data-qmd="As described in @Lovelace1864, computers are great."></span></caption>
<thead>
<tr>
<th><span data-qmd="_Header 1_"></span></th>
<th><span data-qmd="_Header 2_"></span></th>
</tr>
</thead>
<tbody>
<tr>
<td><span data-qmd=""></span></td>
<td>Regular output</td>
</tr>
</tbody>
</table>
```
````

**Reply by cscheid** [reply]
And I realize this is now crashing quarto for me 🤦. But your syntax should 100% work. Let me get back to you asap.

**Reply by baptiste** [reply]
I also tried this but it froze on me, so I was very confused!

**Reply by cscheid** [reply]
The bug I'm about to fix is from `data-qmd=""` not being handled well, fwiw.

**Reply by cscheid** [reply, marked as answer]
````
---
title: "testing table"
format: 
   html:
     keep-md: true
   pdf:
     keep-tex: true
     number-sections: true
---

## Introduction 


```{=html}
<table>
<caption><span data-qmd="As described in @sec-content, computers are great."></span></caption>
<thead>
<tr>
<th><span data-qmd="_Header 1_"></span></th>
<th><span data-qmd="_Header 2_"></span></th>
</tr>
</thead>
<tbody>
<tr>
<td><span data-qmd=""></span></td>
<td>Regular output</td>
</tr>
</tbody>
</table>
```




## Content {#sec-content}

Some content here.
````

On my (fixed) quarto, you'll see this:

<img width="570" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/74978c79-01f7-4545-8645-ec803a257f19">

**Reply by cscheid** [reply]
Waiting for https://github.com/quarto-dev/quarto-cli/commit/16f3cc909d27a5e388be5286dd5f3d6c118dfb18 to pass the test suite and then we'll fire a new release. 1.4.69 should have the fix.

**Reply by baptiste** [reply]
Nice, thanks! Actually, my table doesn't have empty spans so it would have worked fine if I'd tried it within a `{=html}` raw block, but I tried that with the smaller example...

**Reply by cscheid** [reply]
Should be fixed on 1.4.70.
